### ISSUE TITLE
`didArchiveLogFile` returns the file name instead of the file path

### ISSUE THREAD

**Issue by zbencz3** [CocoaLumberjack/CocoaLumberjack#1078]
### New Issue Checklist

* [X] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [X] I have read the [Documentation](https://github.com/CocoaLumberjack/CocoaLumberjack#documentation) or [CocoaDocs](cocoadocs.org/docsets/CocoaLumberjack)
* [X] I have searched for a similar issue in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios
 Platform Version        | 12.2
 CocoaLumberjack Version | 3.5.3
 Integration Method      | cocoapods
 Xcode Version           | 10.2
 Repro rate              | all the time (100%)
 Demo project link     | none

### Issue Description and Steps

The `didArchiveLogFile:` specifies that it returns the file path, `logFilePath`, but it seems to return the file name instead.
See: `NSString *archivedLogFilePath = [logFileInfo.fileName copy];`
```
- (BOOL)lt_shouldUseLogFile:(nonnull DDLogFileInfo *)logFileInfo isResuming:(BOOL)isResuming {
...
   if (isResuming && (_doNotReuseLogFiles || [self lt_shouldLogFileBeArchived:logFileInfo])) {
        logFileInfo.isArchived = YES;
        NSString *archivedLogFilePath = [logFileInfo.fileName copy];

        if ([_logFileManager respondsToSelector:@selector(didArchiveLogFile:)]) {
            dispatch_async(_completionQueue, ^{
                [self->_logFileManager didArchiveLogFile:archivedLogFilePath];
            });
        }

        return NO;
    }
```

Is this the intended behavior? 
This seems to be a change introduced in version `3.5.0`. Version `3.4.2` had the correct file path as expected.

If this needs correction, I am willing to create a pull request of course.

Thanks.

**Comment by ffried** [comment]
@zbencz3 Thanks for reporting this. You're absolutely right, this should be the path of the file instead of just the name. [This commit](62e4ef14908f4c891394dd497b954288e0162d52) introduced this little mistake.
Please do open a PR for this. If you could also add a test for this, it would be even better. 🙂 

In case you need it, here's a little starting ground on how you could test this:
You can simply create a subclass that returns `YES` for `shouldArchiveRecentLogFileInfo:` and then log a message. This should then call into `lt_shouldUseLogFile:isResuming:` and archive the log file. In a subclass of the log file manager you can then store the path passed to `didArchiveLogFile:` and then check it in the test case. Make sure you log an error (`DDLogError`) so that logging happens synchronously.

**Comment by lolgear** [comment]
Or you can wrap any logger into NSProxy and do the same. ( if you use ObjC, of course. )

**Comment by zbencz3** [comment]
@ffried I created a branch for this issue, but when I try to push my changes I get 
`Pushing to https://github.com/CocoaLumberjack/CocoaLumberjack
remote: Permission to CocoaLumberjack/CocoaLumberjack.git denied to zbencz3.
fatal: unable to access 'https://github.com/CocoaLumberjack/CocoaLumberjack/': The requested URL returned error: 403`

I might be missing a step, but could not find it in [contributing guidelines](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md).
How should I proceed?
